aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -0,0 +1,223 @@
+import base64
+import json
+import re
+import zlib
+from enum import Enum
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
+
+from aws_lambda_powertools.shared.json_encoder import Encoder
+from aws_lambda_powertools.utilities.data_classes import ALBEvent, APIGatewayProxyEvent, APIGatewayProxyEventV2
+from aws_lambda_powertools.utilities.data_classes.common import BaseProxyEvent
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+class ProxyEventType(Enum):
+    http_api_v1 = "APIGatewayProxyEvent"
+    http_api_v2 = "APIGatewayProxyEventV2"
+    alb_event = "ALBEvent"
+    api_gateway = http_api_v1
+
+
+class CORSConfig(object):
```

Comment:
Could you add a docstring to quickly note what each CORS option is for to help educate customers who are new to this?

I know many struggle with the origin value.

**Note**: I can do this in the docs PR too, so don't worry about it.